### ISSUE TITLE
Partitioning conflicts exiting partitions

### ISSUE THREAD

**Issue by chumaltd** [ridgepole/ridgepole#385]
# Description

* Commands and arguments when executing Ridgepole
  `ridgepole -c database.yml -f Schemafile -a -Edevelopment`
* What wasn't as expected?
  Not migrated at all.
* Middleware versions
  * Ridgepole 1.0.3
  * Ruby 2.7.0
  * Rails/Active Record 7.0.2
  * PostgreSQL

#374 introduced partial support for table partitioning. It's great!
But it conflicts existing partitions created with options other than supported.

It retrieves configrations from not Schemafile but existing database instances by design.
So, it aborts entire migragtions if you have tables includidng NOT SUPPORTED options.

Cases for PostgreSQL are:

* Current ridgepole accepts only `FROM` and `TO` combination for `RANGE` and `LIST` partitions. It's supported.
* PostgreSQL has partitioning options like `DEFAULT`, `WITH` since v11. They are not suppored by ridgepole current.
  https://www.postgresql.org/docs/11/sql-createtable.html

For my case, I have tables including `DEFAULT` partitions, and ridgepole gets stuck at parsing options.

https://github.com/ridgepole/ridgepole/blob/1.0/lib/ridgepole/ext/postgresql_adapter/partitioning.rb#L61

```
[ERROR] undefined method `[]' for nil:NilClass
        /var/lib/gems/2.7.0/gems/ridgepole-1.0.3/lib/ridgepole/ext/postgresql_adapter/partitioning.rb:61:in `block in partition'
```

I don't know other than `DEFAULT`, but I think it potentially affects diverse installations.

#374 is basically useful, so full support for existing partitioning options looks needed.
Until its completion, some bypassing option is required.

**Comment by chumaltd** [comment]
For now, I just confirmed that Ridgepole v1.0.1 with ActiveRecord v7.0.1 works well.
This is not urgent for me.

Thanks a lot for good works.

**Comment by winebarrel** [comment]
@chumaltd Can you write a Schemafile and DDL that didn't work as expected?

**Comment by chumaltd** [comment]
@winebarrel I just created a schema for check.
I confirmed with ridgepole v1.0.3 & PG v13.

Schemafile is as follows:

```ruby
create_table "counters", id: false do |t|
  t.bigint               "count"
end
```

And I created 3 DDL flavors for convenience.

## [A] Range partition with DEFAULT

This is what I reported.

```sql
CREATE TABLE counters (
    count bigint
)
PARTITION BY RANGE (count);

CREATE TABLE IF NOT EXISTS counters_100 PARTITION of counters FOR VALUES from(1) to (100);
CREATE TABLE IF NOT EXISTS counters_default PARTITION of counters DEFAULT;
```

## [B] Hash partigion

This gets `NotImplementedError`, which looks intentional but inconvenient.

```sql
CREATE TABLE counters (
    count bigint
)
PARTITION BY HASH (count);

CREATE TABLE IF NOT EXISTS counters_even PARTITION of counters FOR VALUES WITH (MODULUS 2, REMAINDER 0);
CREATE TABLE IF NOT EXISTS counters_odd PARTITION of counters FOR VALUES WITH (MODULUS 2, REMAINDER 1);
```

## [C] List partition

With LIST, ridgepole works without errors.
I don't know wheter it is good behavior or not.
DDL is as follows:

```sql
CREATE TABLE counters (
    count bigint
)
PARTITION BY LIST (count);

CREATE TABLE IF NOT EXISTS counters_1 PARTITION of counters FOR VALUES in (1);

-- Actual scenario should have exhaustive list including default.
--
-- CREATE TABLE IF NOT EXISTS counters_default PARTITION of counters DEFAULT;
```

**Comment by winebarrel** [comment]
@chumaltd I fixed it in v1.0.5, please try it.

**Comment by winebarrel** [comment]
@chumaltd v1.0.6 has been released, try it out.
